### PR TITLE
test primitives

### PR DESCRIPTION
**Pull request by svandenb-dev**
@maxcapodi78 this PR is a test nothing finalized. Not sure if you agree I put here. Could you please check if you agree with the approach before I go further ?
-Simon

### CODE REVIEW

File: pyaedt/edb_core/primitives.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+"""
+This module .
+"""
+
+from pyaedt import aedt_exception_handler
+from pyaedt.edb_core.layout import EdbLayout
+from pyaedt.edb_core.padstack import EdbPadstacks
+
+class trace(object):
```
Comment (maxcapodi78) [inline pyaedt/edb_core/primitives.py:9]: use CamelCase for class names
```suggestion
class Trace(object):
```

File: pyaedt/edb_core/primitives.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    @aedt_exception_handler
+    def place_line(self):
+
+        path = self.layout_core.Shape("polygon", points=self.point_list)
+        edb_path = self.layout_core.create_path(path, self.layer, self.width, self.net_name, start_cap_style="Flat",
+                                                   end_cap_style="Flat")
+        return edb_path
+
+class via_def(object):
```
Comment (maxcapodi78) [inline pyaedt/edb_core/primitives.py:30]: Same as before
```suggestion
class ViaDefinition(object):
```

File: pyaedt/edb_core/primitives.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                                          paddiam=self.pad_diam,
+                                          antipaddiam=self.anti_pad_diam,
+                                          startlayer=self.start_layer,
+                                          endlayer=self.stop_layer,
+                                          antipad_shape=self.anti_pad_shape,
+                                          x_size=self.x_size,
+                                          y_size=self.y_size,
+                                          corner_radius=self.corner_rad)
+
+class via_instance(object):
```
Comment (maxcapodi78) [inline pyaedt/edb_core/primitives.py:56]:
```suggestion
class ViaInstance(object):
```

File: pyaedt/edb_core/primitives.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        self.net_name = net_name
+        self.padstack_core = EdbPadstacks(self)
+
+    def place_via(self, viadef=via_def()):
+        edb_padstanck_inst = self.padstack_core.place_padstack(
+            position=self.pos, definition_name=viadef.name, net_name=self.net_name, via_name="", rotation=self.rotation,
+            fromlayer=viadef.start_layer, tolayer=viadef.stop_layer)
+        return  edb_padstanck_inst
+
+class rectangle(object):
```
Comment (maxcapodi78) [inline pyaedt/edb_core/primitives.py:71]:
```suggestion
class Rectangle(object):
```

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]
I would not define a new class for those object but rather add methods to EDBLayout Class

File: pyaedt/edb_core/primitives.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        if point_list is None:
+            point_list = [[0.0, 0.0], [1e-3, 0.0]]
+        self.point_list = point_list
+        self.width = width
+        self.layer = layer
+        self.net_name = net_name
+        self.layout_core = EdbLayout(self)
+
+    @aedt_exception_handler
+    def place_line(self):
```
Comment (beliaev-maksim) [inline pyaedt/edb_core/primitives.py:23]: do you seriously need a class for a single function?

why just not to use a function?

File: pyaedt/edb_core/primitives.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        self.anti_pad_diam = anti_pad_diam
+        self.start_layer = start_layer
+        self.stop_layer = stop_layer
+        self.anti_pad_shape = antipad_shape
+        self.x_size = x_size
+        self.y_size = y_size
+        self.corner_rad = corner_rad
+        self.padstack_core = EdbPadstacks(self)
+
+    def add_via_def_to_edb(self):
```
Comment (beliaev-maksim) [inline pyaedt/edb_core/primitives.py:45]: same here, it is a class with single function

Comment (svandenb-dev) [reply pyaedt/edb_core/primitives.py:45]: no I started a draft and committed by mistake

File: pyaedt/edb_core/primitives.py (hunk 5 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/primitives.py:23]: same was committed by mistake will remove it

Comment (svandenb-dev) [reply pyaedt/edb_core/primitives.py:23]: btw this PR has been dropped and was never planned to be merged. I will cancel it

**Pull request closed without merging**